agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
+   */
+  private boolean transformClass(ClassGen cg) {
     boolean transformed = false;
+    ConstantPoolGen pgen = cg.getConstantPool();
+    debug_transform.log("transformClass: ENTER %s", cg.getClassName());
+    // Loop through each method in the class
+    for (Method method : cg.getMethods()) {
+      MethodGen mg = new MethodGen(method, cg.getClassName(), pgen);
+
+      // Get the instruction list and skip methods with no instructions
+      InstructionList instructionList = mg.getInstructionList();
+      if (instructionList == null) {
+        continue;
+      }
 
-    try {
-      pgen = cg.getConstantPool();
-
-      // Loop through each method in the class
-      Method[] methods = cg.getMethods();
-      for (int i = 0; i < methods.length; i++) {
-        MethodGen mg = new MethodGen(methods[i], cg.getClassName(), pgen);
-
-        // Get the instruction list and skip methods with no instructions
-        InstructionList il = mg.getInstructionList();
-        if (il == null) continue;
-
-        if (debug) out.format("Original code: %s%n", mg.getMethod().getCode());
-
-        instrument_method(methods[i], mg, current_method_map);
-
-        // Remove the Local variable type table attribute (if any).
-        // Evidently, some changes we make require this to be updated, but
-        // without BCEL support, that would be hard to do. Just delete it
-        // for now (since it is optional, and we are unlikely to be used by
-        // a debugger)
-        for (Attribute a : mg.getCodeAttributes()) {
-          if (is_local_variable_type_table(a)) {
-            mg.removeCodeAttribute(a);
-          }
-        }
+      debug_instrument_inst.log(
+          "%n%s.%s original code: %s%n", mg.getClassName(), mg.getName(), mg.getMethod().getCode());
 
-        // Update the instruction list
-        mg.setInstructionList(il);
-        mg.update();
-
-        // Update the max stack and Max Locals
-        mg.setMaxLocals();
-        mg.setMaxStack();
-        mg.update();
-
-        // Update the method in the class
-        cg.replaceMethod(methods[i], mg.getMethod());
-        if (debug) out.format("Modified code: %s%n", mg.getMethod().getCode());
-
-        // verify the new method
-        // StackVer stackver = new StackVer();
-        // VerificationResult vr = stackver.do_stack_ver (mg);
-        // log ("vr for method %s = %s%n", mg.getName(), vr);
-        // if (vr.getStatus() != VerificationResult.VERIFIED_OK) {
-        // System.out.printf ("Warning BCEL Verify failed for method %s: %s",
-        // mg.getName(), vr);
-        // System.out.printf ("Code: %n%s%n", mg.getMethod().getCode());
-        // System.exit(1);
-        // }
+      if (transformMethod(mg, new InstructionFactory(cg), pgen)) {
+        transformed = true;
       }
 
-      cg.update();
-    } catch (Exception e) {
-      out.format("Unexpected exception encountered: " + e);
-      e.printStackTrace();
+      // Update the instruction list
```

Comment:
Why is this done if `transformed` is false?  Should all this be moved into the branch that sets `transformed` to true?